Update portcullis to 1.2.3

ljyanesm opened the pull request.

ljyanesm committed:
Update portcullis to 1.2.3

alalazo commented on var/spack/repos/builtin/packages/portcullis/package.py:
```diff
@@ -10,9 +10,10 @@
-    homepage = "https://github.com/maplesond/portcullis"
-    url      = "https://github.com/maplesond/portcullis/archive/Release-1.1.2.tar.gz"
+    homepage = "https://github.com/ei-corebioinformatics/portcullis"
+    url      = "https://github.com/ei-corebioinformatics/archive/refs/tags/Release-1.2.3.tar.gz"
```
The `url` seems incorrect. The correct one has the form:
```
https://github.com/EI-CoreBioinformatics/portcullis/archive/refs/tags/Release-1.1.2.tar.gz
```

ljyanesm committed:
Update portcullis

Updated url

Included missing boost dependency and removed extra 'build_boost.sh' call not required any longer

alalazo commented on var/spack/repos/builtin/packages/portcullis/package.py:
```diff
@@ -10,14 +10,16 @@
     depends_on('autoconf@2.53:', type='build')
     depends_on('automake@1.11:', type='build')
     depends_on('libtool@2.4.2:',  type='build')
+    depends_on('boost', type='build')
```
```suggestion
    depends_on('boost')
```
In our current dependency model header only libraries are still of `link` type. Roughly `build` and `run` identify dependencies on executables that are used in a separate process at either build-time or run-time.

alalazo reviewed (changes requested):
Only a minor request

ljyanesm committed:
Remove type argument for boost dependency

Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

ljyanesm commented:
@alalazo, thank you for reviewing these changes

alalazo reviewed (approved):
Thanks!
